cypress/TestCases/FireboltCertification/Manage/UserGrantsManage.feature

```diff
@@ -0,0 +1,190 @@
+
+    # First allowing the grant and calling usergrants.capability then checking grant using Capabilities.granted
+    @Usergrants @manageSDK @sdk @transport @requiresPlatformImplementation
+    Scenario: Usergrants.capability R.5.1.4 - Positive Scenario: with grant(ackchallenge)
+        Given the environment has been set up for 'userGrants' tests
+        And 3rd party 'certification' app is launched
+        And Framework registers 'ackchallenge' test provider
+        And User 'starts' recording lifecycle history for '3rd party app'
+        When '3rd party app' invokes the 'Firebolt' API to 'fetch device id'
+        And User 'stops' recording lifecycle history for '3rd party app'
+        Then User validates lifecycle history for '3rd party app' with 'background:foreground'
+        When 1st party app invokes the 'Firebolt' API to 'get usergrants for device id capability'
+        Then 'Firebolt' platform responds to '1st party app' with 'expected list of grants for device id capability'
+        When '3rd party app' invokes the 'Firebolt' API to 'check if capabilities is granted for device id'
+        Then 'Firebolt' platform responds with 'true for capabilities granted'
+
+    @Usergrants @manageSDK @sdk @transport @requiresPlatformImplementation
+    Scenario Outline: Usergrants.<method> - Positive Scenario: <Scenario>
+        Given the environment has been set up for 'userGrants' tests
+        And 3rd party 'certification' app is launched
+        And 1st party app invokes the 'Firebolt' API to '<Grant_Key>'
+        Then 'Firebolt' platform responds to '1st party app' with '<Grant_Method_Content>'
+        When 1st party app invokes the 'Firebolt' API to '<Key>'
+        Then 'Firebolt' platform responds to '1st party app' with '<Validation_Key>'
+
+        Examples:
+            | Scenario                         | Grant_Key                           | Grant_Method_Content      | Key                              | Validation_Key                                   | Method        |
+            | grant capability at device level | grant device distributor capability | null for usergrants grant | get usergrants for device        | expected list of grants for device               | grant R.5.1.5 |
+            | grant capability at app level    | grant device id capability          | null for usergrants grant | get usergrants for 3rd party app | expected list of grants for 3rd party app        | grant R.5.1.5 |
+            | deny capability at device level  | deny device distributor capability  | null for usergrants deny  | get usergrants for device        | expected list of denied grants for device        | deny R.5.1.6  |
+            | deny capability at app level     | deny device id capability           | null for usergrants deny  | get usergrants for 3rd party app | expected list of denied grants for 3rd party app | deny R.5.1.6  |
+
```

Comment:
what does each column mean ? I am not able to understand what method is being called at the backend ?

Reply:
updated